internal/slicer/slicer.go

```diff
@@ -69,13 +71,48 @@ func Run(options *RunOptions) error {
 		targetDirAbs = filepath.Join(dir, targetDir)
 	}
 
+	// Archives are ordered in descending order by priority. The
+	// default priority is 0 and the maximum allowed priority is
+	// the upper limit of int.
+	//
+	// If a package to be downloaded exists in archive A and
+	// archive B, and archive A has higher priority than archive
+	// B, the package is downloaded from archive A even if archive
+	// B has a newer version.
+	//
+	// If two or more archives have the same priority, the most
+	// recent version of the package from all of them is selected,
+	// if any. Otherwise, archives with lower priority are tried,
+	// if any.
+	//
+	// If two or more archives with the same priority contain the
+	// package with the same version, the first archive from the
+	// options.Archives iterator is used. In effect, the choice is
+	// random.
+	orderedArchives := make([]archive.Archive, 0, len(options.Archives))
+	for _, archive := range options.Archives {
+		orderedArchives = append(orderedArchives, archive)
+	}
+	sort.Slice(orderedArchives, func(a, b int) bool {
+		prioA := orderedArchives[a].Options().Priority
+		prioB := orderedArchives[b].Options().Priority
+		return prioA > prioB
+	})
+
 	// Build information to process the selection.
 	for _, slice := range options.Selection.Slices {
 		extractPackage := extract[slice.Package]
 		if extractPackage == nil {
 			var selectedVersion string
 			var selectedArchive archive.Archive
-			for _, archive := range options.Archives {
+			currentPrio := math.MaxInt
```

Comment:
If the priorities are written in `chisel.yaml` with 64-bit ints, this might fail while running on a 32-bit arch.

Reply:
Nice catch. Fixed in new commit `Make archive priority int32`
```
    Archive priority is currently defined as int. This can make chisel.yaml
    with very high/low priorities non-portable, e.g. from amd64 to arm32.
    
    Change archive priority definition to int32 data type.
```